ComputedGoto: bad codegen with enum with holes

Taking the test here: https://github.com/nim-lang/Nim/blob/ddc131cf07972decc206e033b2dd85a42eb1c98d/tests/casestmt/tcomputedgoto.nim and changing the enum to enum with holes breaks codegen:

```Nim
discard """
  output: '''yeah A enumB
yeah A enumB
yeah CD enumD
yeah CD enumE
yeah A enumB
yeah CD enumE
yeah CD enumD
yeah A enumB
yeah B enumC
yeah A enumB
yeah A enumB
yeah A enumB'''
"""

type
  MyEnum = enum
    enumA = 0x0000
    enumB = 0x0100
    enumC = 0x0101
    enumD = 0x0200
    enumE = 0x0300
    enumLast = 0x0900

proc vm() =
  var instructions: array[0..100, MyEnum]
  instructions[2] = enumC
  instructions[3] = enumD
  instructions[4] = enumA
  instructions[5] = enumD
  instructions[6] = enumC
  instructions[7] = enumA
  instructions[8] = enumB

  instructions[12] = enumE
  var pc = 0
  while true:
    {.computedGoto.}
    let instr = instructions[pc]
    let ra = instr.succ # instr.regA
    case instr
    of enumA:
      echo "yeah A ", ra
    of enumC, enumD:
      echo "yeah CD ", ra
    of enumB:
      echo "yeah B ", ra
    of enumE:
      break
    of enumLast: discard
    inc(pc)

vm()
```

```Nim
.../nimcache/compgoto.c:92:21616: error: use of undeclared label 'TMP1894_'
.../nimcache/compgoto.c:92:26536: error: use of undeclared label 'TMP2304_'
.../nimcache/compgoto.c:92:2187: error: use of undeclared label 'TMP209_'
.../nimcache/compgoto.c:92:14344: error: use of undeclared label 'TMP1288_'
.../nimcache/compgoto.c:92:6004: error: use of undeclared label 'TMP556_'
.../nimcache/compgoto.c:92:26524: error: use of undeclared label 'TMP2303_'
.../nimcache/compgoto.c:92:2198: error: use of undeclared label 'TMP210_'
.../nimcache/compgoto.c:92:14356: error: use of undeclared label 'TMP1289_'
.../nimcache/compgoto.c:92:5993: error: use of undeclared label 'TMP555_'
.../nimcache/compgoto.c:92:14368: error: use of undeclared label 'TMP1290_'
.../nimcache/compgoto.c:92:21604: error: use of undeclared label 'TMP1893_'
.../nimcache/compgoto.c:92:26512: error: use of undeclared label 'TMP2302_'
.../nimcache/compgoto.c:92:5971: error: use of undeclared label 'TMP553_'
.../nimcache/compgoto.c:92:26560: error: use of undeclared label 'TMP2306_'
.../nimcache/compgoto.c:92:5982: error: use of undeclared label 'TMP554_'
.../nimcache/compgoto.c:92:2154: error: use of undeclared label 'TMP206_'
.../nimcache/compgoto.c:92:14380: error: use of undeclared label 'TMP1291_'
.../nimcache/compgoto.c:92:14392: error: use of undeclared label 'TMP1292_'
.../nimcache/compgoto.c:92:2176: error: use of undeclared label 'TMP208_'
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

--- a/compiler/ccgstmts.nim
+++ b/compiler/ccgstmts.nim
@@ -404,6 +404,9 @@ proc genComputedGoto(p: BProc; n: PNode) =
         localError(p.config, it.info,
             "case statement must be exhaustive for computed goto"); return
       casePos = i
+      if enumHasHoles(it.sons[0].typ):
+        localError(p.config, it.info,
+            "case statement cannot work on enums with holes for computed goto"); return
       let aSize = lengthOrd(p.config, it.sons[0].typ)
       if aSize > 10_000:
         localError(p.config, it.info,

--- a/tests/casestmt/t7699.nim
+++ b/tests/casestmt/t7699.nim
@@ -1,0 +1,15 @@
+discard """
+  line: 13
+  errormsg: "case statement cannot work on enums with holes for computed goto"
+"""
+
+type
+  X = enum
+    A = 0, B = 100
+
+var z = A
+while true:
+  {.computedGoto.}
+  case z
+  of A: discard
+  of B: discard

